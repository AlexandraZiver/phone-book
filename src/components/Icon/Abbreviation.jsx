import PropTypes from "prop-types";
import React from "react";

const Abbreviation = ({ client }) => {
  const showAbbrev = (name) => {
    if (name) {
      return name[0].toUpperCase();
    }
  };

  return (
    <div>
      {showAbbrev(client.general.firstName)}
      {showAbbrev(client.general.lastName)}
    </div>
  );
};

Abbreviation.propTypes = {
  client: PropTypes.shape({
    general: PropTypes.shape({
      firstName: PropTypes.string,
      lastName: PropTypes.string,
      avatar: PropTypes.string,
    }),
    job: PropTypes.shape({
      company: PropTypes.string,
      title: PropTypes.string,
    }),
    contact: PropTypes.shape({
      email: PropTypes.string,
      phone: PropTypes.string,
    }),
    address: PropTypes.shape({
      street: PropTypes.string,
      city: PropTypes.string,
      zipCode: PropTypes.string,
      country: PropTypes.string,
    }),
  }).isRequired,
};

export default Abbreviation;
