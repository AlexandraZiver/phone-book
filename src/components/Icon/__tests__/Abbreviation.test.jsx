import { render, screen } from "@testing-library/react";

import Abbreviation from "../Abbreviation";

describe("Abbreviation testing", () => {
  test("renders the abbreviation correctly", () => {
    render(<Abbreviation firstName="John" lastName="Doe" />);

    const abbreviation = screen.getByText("JD");

    expect(abbreviation).toBeInTheDocument();
    expect(abbreviation).toMatchSnapshot();
  });

  test("render an abbreviation from one word ", () => {
    render(<Abbreviation firstName="John" />);

    const abbreviation = screen.getByText("J");

    expect(abbreviation).toBeInTheDocument();
    expect(abbreviation).toMatchSnapshot();
  });
});
