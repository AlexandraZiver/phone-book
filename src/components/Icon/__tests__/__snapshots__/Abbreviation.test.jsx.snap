// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abbreviation testing render an abbreviation from one word  1`] = `
<div>
  J
</div>
`;

exports[`Abbreviation testing renders the abbreviation correctly 1`] = `
<div>
  JD
</div>
`;
